hbase-client/src/main/java/org/apache/hadoop/hbase/client/ZKConnectionRegistry.java

```diff
@@ -138,8 +141,15 @@ private void getMetaRegionLocation(CompletableFuture<RegionLocations> future,
     if (metaReplicaZNodes.isEmpty()) {
       future.completeExceptionally(new IOException("No meta znode available"));
     }
-    HRegionLocation[] locs = new HRegionLocation[metaReplicaZNodes.size()];
-    MutableInt remaining = new MutableInt(locs.length);
+    // Note, the list of metaReplicaZNodes may be discontiguous regards replicaId; i.e. we may have
+    // a znode for the default -- replicaId=0 -- and perhaps replicaId '2' but be could be missing
+    // znode for replicaId '1'. This is a transient condition. Because of this we are careful
+    // accumulating locations. We use a Map so retries overwrite rather than aggregate and the
+    // Map sorts just to be kind to further processing. The Map will retain the discontinuity on
+    // replicaIds but on completion (of the future), the Map values are passed to the
+    // RegionLocations constructor which knows how to deal with discontinuities.
+    final Map<Integer, HRegionLocation> locs = new TreeMap();
```

Comment:
This will generate a generic type warning, please use `new TreeMap<>()`

Reply:
Fixed